src/index.ts

```diff
@@ -29,66 +32,53 @@ if ('serviceWorker' in navigator && process.env.NODE_ENV !== 'development') {
 
 window.addEventListener('load', () => {
   const fileInput = document.querySelector('#input') as HTMLElement;
-  const imgElement = document.querySelector('img') as HTMLImageElement;
-  const shareBtn = document.querySelector('#share') as HTMLButtonElement;
+  const mainElement = document.querySelector('main') as HTMLElement;
 
   if (!window.indexedDB || !window.URL) { /* TODO: display error message */ }
 
   const dbPromise = LlamaStorage.create();
-  dbPromise.then((db) => {
-    shareBtn.addEventListener('click', async () => {
-      const id = await db.getCurrentID();
-      if (!id) return; // TODO: show error message
-      const file = await fileFromID(await dbPromise, id);
-      if (!file) return; // TODO: show error message
-      navigator.share({files: [file]});
-    });
-    onDBOpenSuccess(db, imgElement, shareBtn);
+  dbPromise.then(async db => {
+    const id = await db.getCurrentID();
+    if (!id) return;
+
+    const blob = await db.getFile(id);
+    if (!blob) return;
+
+    createOrReplaceVisibleItem(mainElement, dbPromise, id, URL.createObjectURL(blob));
   });
 
-  fileInput.addEventListener('fileselected', (e) => onFileInputChange(e as CustomEvent, dbPromise, imgElement, shareBtn));
+  fileInput.addEventListener('fileselected', e =>
+      onFileInputChange(e as CustomEvent, mainElement, dbPromise));
 });
 
-async function onDBOpenSuccess(db: LlamaStorage, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
-  const id = await db.getCurrentID();
-  if (!id) return;
-  const blob = await db.getFile(id);
-  if (!blob) return;
-  imgElement.src = window.URL.createObjectURL(blob);
-  displayIfShareEnabled(shareBtn, db);
+function createOrReplaceVisibleItem(container: HTMLElement,
+                                    dbPromise: Promise<LlamaStorage>,
+                                    id: string,
+                                    src: string): void {
+  // TODO: Support displaying any number of stored items in Llaminator. For now,
+  // remove all existing content from the |container| to ensure that only a
+  // single item is displayed.
+  while (container.children.length)
+    container.firstChild?.remove();
+
+  render(html`
+    <llama-item .storage=${dbPromise} id=${id} src=${src}>
```

Comment:
why is this `.storage` rather than `storage`?

Reply:
HTML attributes are strings. In this instance we are setting a _property_, which uses a different communication channel in lit-element to allow for richer values. The prefixing period indicates that.